File: 73-set-matrix-zeroes/set-matrix-zeroes.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        m=len(matrix)
        n=len(matrix[0])
        col=[0]*m
        row=[0]*n
        for i in range(m):
            for j in range(n):
                if matrix[i][j]==0:
                    col[i]=1
                    row[j]=1
        for i in range(m):
            for j in range(n):
                if col[i]==1 or row[j]==1:
                    matrix[i][j]=0

File: 73-set-matrix-zeroes/set-matrix-zeroes.js
/**
 * @param {number[][]} matrix
 * @return {void} Do not return anything, modify matrix in-place instead.
 */
var setZeroes = function(matrix) {
    let m = matrix.length;
    let n = matrix[0].length;
    console.log(m,n);
    let cols = new Array(m);
    let rows = new Array(n);
    for(let i=0; i<m; i++){
        for(let j=0; j<n; j++){
            if(matrix[i][j]==0){
                cols[i]=1;
                rows[j]=1;
            };
        };
    };
    for(let i=0; i<m; i++){
        for(let j=0; j<n; j++){
            if(cols[i]==1 || rows[j]==1){
                matrix[i][j]=0;
            };
        };
    };
};
